### ISSUE TITLE
DateTimePicker and DatePicker value change listener get fired twice when setting value with setValue() method.

### ISSUE THREAD

**Issue by paodb** [vaadin/flow-components#1986]
### Description
When setting a value for DateTimePicker or DatePicker using setValue() method, the value change listener gets fire twice.
When loading  the view the first time (or refreshing the page) one event is reported to come from the server, and one from the client, when there has been no client interactions to the field, just a setValue() on the server. 
When calling setValue(), the value change listener fires two times, but at least both of them are reported as coming from the server.

### Expected outcome

Value change listener to fire once and with event from the right source.

### Actual outcome

Value change listener is being fired twice and events are not always from server.

### Minimal reproducible example

```  
 public MainView() {
        DateTimePicker dateTimePicker = new DateTimePicker();
        dateTimePicker.addValueChangeListener(event -> {
          if (event.isFromClient()) {
              System.out.println("DateTimePicker event is from client!");
          } else {
              System.out.println("DateTimePicker event is from server!");
          }        
        });
        add(dateTimePicker);
        dateTimePicker.setValue(LocalDateTime.now());
        Button setDateTime = new Button("Set date time");
        setDateTime.addClickListener(event -> {
          System.out.println("Button clicked");
          dateTimePicker.setValue(LocalDateTime.now().minusDays(5));
        });        
        add(setDateTime);
    }
```

### Steps to reproduce

1. Add a DateTimePicker component to a page with no initial value.
2. Add a ValueChangeListener to the DateTimePicker component to check if fired event is from client or from server.
3. Add a button to the same page, to set a value to the DateTimePicker component on button click.
4. On page initializatin (or by refreshing) check console to see fired events and where the come from.
5. Click button and check console to see fired events.

### Environment

Vaadin platform 14.6.7.

### Browsers Affected
Tested on Chrome.

**Comment by simasch** [comment]
It's the same with DataPicker

**Comment by sissbruecker** [comment]
@paodb I would like some clarification. There are two problems here:
1. Setting a value from the server-side triggers a value change event that comes from the client
2. Setting a value from the server-side can trigger two value change events that come from the server

(1) Is definitely a bug that I would try to address as part of this issue.
(2) Is an existing workaround for a separate issue

Can you please clarify which of these two issues (or both) you would expect to be fixed as part of this issue?

Also regarding this:
> When calling setValue(), the value change listener fires two times, but at least both of them are reported as coming from the server.

By this you mean when calling `setValue` for the second time?

**Comment by paodb** [comment]
Hello @sissbruecker, I think both issues are important but main issue to get fix will be number 1, as this ticket was on respond of getting confused when trying to know if value of the component was coming from client or server side.

Both issues seems to be fixed with the provided workaround. (I just did a quick test adding that to the reproducible example I added on description, using Vaadin version 14.6.8).

Regarding your last question, when calling setValue() from button click in the example I notice this behavior.

@simasch do you agree?

**Comment by simasch** [comment]
@sissbruecker and @paodb 
I agree. The workaround fixes the issue for us.

**Comment by ZheSun88** [comment]
This ticket/PR has been released with platform 22.0.0.alpha4 and is also targeting the upcoming stable 22.0.0 version.

**Comment by ZheSun88** [comment]
This ticket/PR has been released with platform 21.0.2.

**Comment by ZheSun88** [comment]
This ticket/PR has been released with platform 14.7.1.
